KarhooUISDK/Screens/BookingRequestScreen/FormBookingRequestScreen/FormBookingRequestViewController.swift

```diff
@@ -294,60 +434,96 @@ final class FormBookingRequestViewController: UIViewController, BookingRequestVi
     
     func setAddFlightDetailsState() {
         enableUserInteraction()
-        poiDetailsInputText.isHidden = false
+        if presenter.isKarhooUser() {
+            bookingButton.setAddFlightDetailsMode()
+        } else {
+            poiDetailsInputText.isHidden = false
+        }
     }
     
     func set(quote: Quote) {
-        let viewModel = QuoteViewModel(quote: quote)
-        headerView.set(viewModel: viewModel)
-        termsConditionsView.setBookingTerms(supplier: quote.fleet.name,
-                                            termsStringURL: quote.fleet.termsConditionsUrl)
-        addPaymentView.quote = quote
+        switch Karhoo.configuration.authenticationMethod() {
```

Comment:
I think we only need
```
    func set(quote: Quote) {
        let viewModel = QuoteViewModel(quote: quote)
        headerView.set(viewModel: viewModel)
        termsConditionsView.setBookingTerms(supplier: quote.fleet.name,
                                            termsStringURL: quote.fleet.termsConditionsUrl)
        addPaymentView.quote = quote
    }
```

Reply:
I've refactored this method to this: 
`    func set(quote: Quote) {
        let viewModel = QuoteViewModel(quote: quote)
        headerView.set(viewModel: viewModel)
        termsConditionsView.setBookingTerms(supplier: quote.fleet.name,
                                            termsStringURL: quote.fleet.termsConditionsUrl)
        switch Karhoo.configuration.authenticationMethod() {
        case .karhooUser:
            paymentView.quote = quote
        default:
            addPaymentView.quote = quote
        }
    }`
I removed the supplierView and used headerView, but the switch is still there for populating either paymentView or the addPaymentView